hedera-node/src/test/java/com/hedera/services/legacy/unit/handler/FileServiceHandler.java

```diff
@@ -115,6 +121,115 @@ public static FileInfo lookupInfo(FileID fid, FCStorageWrapper fcfs) throws Exce
 		}
 	}
 
+  /**
+   * Checks if an account has authority to update an entity based on the following rules: Treasury
+   * account can update all entities below 0.0.1001 Account 0.0.50 can update all entities from
+   * 0.0.50 - 0.0.80 Network Function Master Account A/c 0.0.50 - Update all Network Function
+   * accounts & perform all the Network Functions listed below Network Function Accounts
+   * A/c 0.0.55 - Update Address Book files (0.0.101/102) + Properties/Permission files (0.0.121/122)
+   * A/c 0.0.56 - Update Fee schedule (0.0.111)
+   * A/c 0.0.57 - Update Exchange Rate (0.0.112) + Properties/Permission files (0.0.121/122)
+   *
+   * @param account account to be checked
+   * @param entity entity to be updated
+   * @return true if the account has the authority to update the entity, false otherwise
+   */
+  public static boolean hasAuthorityToUpdate(AccountID account, Object entity) {
+    if (isTreasury(account)) {
+      return true;
+    }
+
+    boolean rv = false;
+    long seq = getEntityNumber(entity);
+    if (isMasterAccount(account)) {
+      if((entity instanceof AccountID) &&
+          isMasterAccount((AccountID) entity)) { // master account cannot update itself
+        return false;
+      }
+      else if (seq >= 50 && seq <= 80)
+        rv = true;
+      else if(seq == 101 || seq == 102 || seq == 111 || seq == 112
```

Comment:
`FileServiceHandler` is essentially "to-be-deleted"; it remains because the legacy Smart Contract tests use it. We can pretend it doesn't exist for the purposes of this PR tho. 😁

Reply:
Ah Understood. Thanks !! :D